lib/internal/util/inspect.js

```diff
@@ -639,12 +639,15 @@ function formatRaw(ctx, value, recurseTimes) {
       base = `${regExpToString(value)}`;
     } else if (isDate(value)) {
       // Make dates with properties first say the date
+      base = Number.isNaN(dateGetTime(value)) ?
+        keys.length === 0 && String(value) :
```

Comment:
Sorry, I didn't get this.. I feel `keys.length` is required, in case we didn't check for it, we will be getting an error (on null prototypes) `String(Object.create(null))` throws:

```
Uncaught TypeError: Cannot convert object to primitive value
```

Reply:
Ah, we have to use the original dates toString function:

```js
base = Number.isNaN(dateGetTime(value)) ?
  dateToString(value) :
  dateToISOString(value);

// At the top of the file:
const dateToString = uncurryThis(Date.prototype.toString);
```